ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FooSubmission.scala

```diff
@@ -8,36 +8,53 @@ import com.daml.ledger.client.binding
 
 import scala.concurrent.{ExecutionContext, Future}
 
-/** Generates and submits Foo and related commands
-  */
 class FooSubmission(
     submitter: CommandSubmitter,
     maxInFlightCommands: Int,
     submissionBatchSize: Int,
     submissionConfig: FooSubmissionConfig,
     signatory: binding.Primitive.Party,
     allObservers: List[binding.Primitive.Party],
+    allDivulgees: List[binding.Primitive.Party],
 ) {
 
   def performSubmission()(implicit
       ec: ExecutionContext
   ): Future[Unit] = {
-    val generator = new FooCommandGenerator(
-      randomnessProvider = RandomnessProvider.Default,
-      signatory = signatory,
-      config = submissionConfig,
-      allObservers = allObservers,
-    )
+    val divulgenceGenerator = new FooDivulgerCommandGenerator()
+    val (divulgerCmds, divulgeesToDivulgerKeyMap) = divulgenceGenerator
+      .makeCreateDivulgerCommands(
+        divulgingParty = signatory,
+        allDivulgees = allDivulgees,
+      )
+
     for {
+      _ <-
+        if (divulgeesToDivulgerKeyMap.nonEmpty) {
```

Comment:
Although you could test on the map or the commands the commands would seem more obvious as it is what is used on the command ?

Reply:
Changing